INSTALL.rst

```diff
@@ -219,9 +245,12 @@ In order to build the documentation you must have additionally
 ``pandoc`` installed. Use your system’s package manager to do this
 (e.g. ``sudo apt-get install pandoc``).
 
-You can build the entire documentation simply by using sphinx from
-``setup.py``:
+You can build the entire documentation simply by using the ``generate_docs.sh``
+script:
 
 .. code:: sh
 
-   python3 setup.py build_ext --inplace docs
+   bash docs/generate_docs.sh
```

Comment:
This is incorrect. I cloned the repo and ran the command:
```
$ bash docs/generate_docs.sh
+++ dirname docs/generate_docs.sh
++ realpath docs
+ script_dir=/home/lucg/sandbox/nmodl/pr-1147/nmodl/docs
+ cd /home/lucg/sandbox/nmodl/pr-1147/nmodl/docs
+ doxygen Doxyfile
error: configuration file Doxyfile not found!
```

Reply:
Ah, good catch! Apparently we use CMake to generate it:
https://github.com/BlueBrain/nmodl/blob/25eb4a055fee4ba7a2e77a621b36834f9d028312/CMakeLists.txt#L194
so I guess we can just use `Doxygen.in` instead of `Doxygen`, and get rid of the useless invocation in `CMakeLists.txt`? Note that the CI doesn't pick this up because `pip wheel . --no-deps` runs CMake

Reply:
I think lines like this:
https://github.com/BlueBrain/nmodl/blob/25eb4a055fee4ba7a2e77a621b36834f9d028312/docs/Doxyfile.in#L867

are why you can't just use `Doxygen.in`. I was expecting this to also configure the version for Doxygen.

Reply:
Ah, so in order to document the project, we first need to build it? Should the documentation then run `pip wheel . --no-deps` to get around this? Nothing else really comes to mind...

Reply:
Try not running `doxygen` manually. It should be taken care of here:
```
https://github.com/BlueBrain/nmodl/blob/fbf9de61d02b252a1c8da54ef3c7e98175920cdf/docs/conf.py#L36
```

Reply:
Now `bash docs/generate_docs.sh` should work more or less without issues, and you can also specify a build directory (so you can reuse it) using `bash docs/generate_docs.sh DIRECTORY` (it defaults to a temporary dir).

Reply:
It doesn't work on my machine:
```
+ sphinx-build . /home/lucg/sandbox/nmodl/pr-1147/nmodl/docs/../public
Running Sphinx v7.2.6

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/sphinx/config.py", line 358, in eval_config_file
    exec(code, namespace)  # NoQA: S102
    ^^^^^^^^^^^^^^^^^^^^^
  File "/home/lucg/sandbox/nmodl/pr-1147/nmodl/docs/conf.py", line 21, in <module>
    import nmodl  # isort:skip
    ^^^^^^^^^^^^
ModuleNotFoundError: No module named 'nmodl'
```

Reply:
Ahh sorry, I forgot about the state of my venv so it worked locally... For some reason we import NMODL in `conf.py`, so we need to first build the module. I think we can just mimic the behavior of `packaging/test_wheels.bash` and install it in a separate venv by default (with the option to do it in the current env), then build the docs using the module just installed.